### PR TITLE
Ut.2-ut.3 Create unit tests for crud operations on product entity, add H2 database, configure liquibase scripts

### PR DESCRIPTION
**Pull request by maxturko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -162,6 +168,17 @@
 							<changeLogFile>db\migration\dbchangelog.xml</changeLogFile>
 						</configuration>
 					</execution>
+					<execution>
+						<id>liquibasetest</id>
```
Comment (DanwerOK) [inline pom.xml]: test

File: pom.xml [hunk 2]
```diff
@@ -162,6 +168,17 @@
...
+						<goals>
+							<goal>update</goal>
+						</goals>
+						<phase>generate-test-resources</phase>
+						<configuration>
+							<propertyFile>src/test/resources/liquibase.properties</propertyFile>
+							<changeLogFile>db\migration\dbchangelog.xml</changeLogFile>
```
Comment (DanwerOK) [inline pom.xml]: Please use forward slash

File: pom.xml [hunk 3]
```diff
@@ -162,6 +168,17 @@
 							<changeLogFile>db\migration\dbchangelog.xml</changeLogFile>
```
Comment (DanwerOK) [inline pom.xml]: Please use forward slash

File: src/main/java/com/mturko/bestshop/controller/ProductController.java [hunk 4]
```diff
@@ -28,6 +28,9 @@ public ProductController(ProductService productService) {
     @PostMapping
     @ResponseStatus(value = HttpStatus.CREATED)
     public Product addProduct(@RequestBody ProductDto productDto) {
-        return productService.addProduct(productDto);
+        Product product = new Product();
+        product.setName(productDto.getName());
+        product.setPrice(productDto.getPrice());
+        return productService.createProduct(product);
```
Comment (DanwerOK) [inline src/main/java/com/mturko/bestshop/controller/ProductController.java:34]: Please move conversion logic into converter class or at least create separate transformer logic.
See: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/convert/ConversionService.html

File: src/test/java/com/mturko/bestshop/ProductServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    void testSpringContextLoadForJunitTests() {
+        assertNotNull(productService);
+    }
+
+    @BeforeAll
+    static void init(){
+        Product product = new Product();
+        product.setName(productInitialName);
+        product.setPrice(productInitialPrice);
+        testProduct = product;
```
Comment (DanwerOK) [inline src/test/java/com/mturko/bestshop/ProductServiceTest.java]: testProduct = new Product();
        testProduct.setName(productInitialName);
        testProduct.setPrice(productInitialPrice);

File: src/test/java/com/mturko/bestshop/ProductServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    @Order(1)
```
Comment (DanwerOK) [inline src/test/java/com/mturko/bestshop/ProductServiceTest.java]: Tests must be isolated

File: src/main/java/com/mturko/bestshop/controller/ProductController.java (hunk 4 above)
Comment (maxturko) [reply src/main/java/com/mturko/bestshop/controller/ProductController.java:34]: will be done in scope of web.4

File: pom.xml (hunk 2 above)
Comment (maxturko) [reply pom.xml]: done

File: pom.xml (hunk 1 above)
Comment (maxturko) [reply pom.xml]: fixed

File: pom.xml (hunk 3 above)
Comment (DanwerOK) [reply pom.xml]: still relevant

File: src/main/resources/db/migration/1_createschema.xml [hunk 7]
```diff
@@ -44,6 +42,8 @@
             <column name="valid_to" type="varchar(20)"/>
         </createTable>
         <addNotNullConstraint columnDataType="BIGINT(20)" columnName="id" tableName="product" />
+        <addNotNullConstraint columnDataType="varchar(50)" columnName="name" tableName="product" />
+        <addPrimaryKey tableName="product" columnNames="id, name" constraintName="pk_product"/>
```
Comment (DanwerOK) [inline src/main/resources/db/migration/1_createschema.xml]: Please revert this change. If you want to make some attiribute unique, just add unique constraint. After change primary key to group of attributes you allowed following behavior:
id name
1  a
1  b
2  a
It looks strange, isn't it?)))

Here is short article describing the difference: https://www.geeksforgeeks.org/difference-between-primary-key-and-unique-key/

File: src/test/java/com/mturko/bestshop/ProductServiceTest.java [hunk 8]
```diff
@@ -9,69 +9,61 @@
...
-    @BeforeAll
-    static void init(){
-        Product product = new Product();
-        product.setName(productInitialName);
-        product.setPrice(productInitialPrice);
-        testProduct = product;
+    @BeforeEach
+    void setup() {
+        Product emptyProduct = productService.getProductByName(productInitialName);
+        Assertions.assertNull(emptyProduct);
```
Comment (DanwerOK) [inline src/test/java/com/mturko/bestshop/ProductServiceTest.java:27]: What is purpose of this assertion?

File: src/main/resources/db/migration/1_createschema.xml (hunk 7 above)
Comment (maxturko) [reply src/main/resources/db/migration/1_createschema.xml]: Agree, my mistake, changed to unique key

File: src/test/java/com/mturko/bestshop/ProductServiceTest.java (hunk 8 above)
Comment (maxturko) [reply src/test/java/com/mturko/bestshop/ProductServiceTest.java:27]: to make sure we don't have this product created before tests

Comment (DanwerOK) [reply src/test/java/com/mturko/bestshop/ProductServiceTest.java:27]: Up to you :-)

**Review by DanwerOK — APPROVED** [review]
